### PR TITLE
feature: HPE Comware Processor Discovery

### PR DESCRIPTION
**Pull request by Rosiak**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

**Comment by murrant** [comment]
@Rosiak I was wondering if we could use the entPhysicalData in the database, soooo, I re-wrote this.  Do you like this better?
```

if ($device['os'] == 'comware') {
    echo 'HPE Comware ';

    if (dbFetchCell('SELECT COUNT(`entPhysical_id`) FROM `entPhysical` WHERE `device_id` = ?', array($device['device_id']))) {
        $indexes = dbFetchColumn('SELECT `entPhysicalIndex` FROM `entPhysical` WHERE device_id = ? AND entPhysicalName = ?', array($device['device_id'], 'Board'));
    } else {
        $entity_data = snmpwalk_cache_oid($device, 'entPhysicalName', array(), 'ENTITY-MIB');
        $indexes = array_keys(array_filter($entity_data, function ($entry) {
            return $entry['entPhysicalName'] == 'Board';
        }));
    }

    if (!empty($indexes)) {
        $procdata = snmpwalk_cache_oid($device, 'hh3cEntityExtCpuUsage', array(), 'HH3C-ENTITY-EXT-MIB');

        foreach ($indexes as $i => $entIndex) {
            if (isset($procdata[$entIndex])) {
                $cur_oid = ".1.3.6.1.4.1.25506.2.6.1.1.1.1.6.$entIndex";
                $cur_value = $procdata[$entIndex]['hh3cEntityExtCpuUsage'];
                $descr = 'Slot ' . ++$i;
                discover_processor($valid['processor'], $device, $cur_oid, $entIndex, 'comware', $descr, '1', $cur_value);
            }
        }
    }
}
```

File: includes/discovery/processors/comware.inc.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    if (!empty($entphydata)) {
+        $procdata = snmpwalk_cache_multi_oid($device, 'hh3cEntityExtCpuUsage', array(), 'HH3C-ENTITY-EXT-MIB');
+        foreach ($entphydata as $index => $phyvalue) {
+            if (stristr($phyvalue['entPhysicalName'], 'Board')) {
+                foreach ($procdata as $procindex => $value) {
+                    if ($index == $procindex) {
+                        $cur_oid = '.1.3.6.1.4.1.25506.2.6.1.1.1.1.6.';
+
+                        discover_processor($valid['processor'], $device, $cur_oid . $procindex, $procindex, 'comware', 'Slot ' . $x, '1', $value['h3cEntityExtCpuUsage']);
```
Comment (murrant) [inline includes/discovery/processors/comware.inc.php]: Should be hh3cEntityExtCpuUsage

**Comment by Rosiak** [comment]
Perfectly fine with that @murrant !

**Comment by laf** [comment]
lol, I'm against that personally as per the other PR which did the same.

Doing it this way requires entity-physical discovery module to be enabled and have run before this (admittedly after the second run it won't matter if entity-physical runs first or not). However the dependancy on another module imho is a no no as you're restricting users who have sensors enabled because they want to see them but not entity-physical.

Other discovery modules which require entityphysical info do another walk afaik - which if we can get the updated snmp Class finished and merged won't be an issue as the results will be cached.

**Comment by murrant** [comment]
@laf I specifically made a fallback so this doesn't rely on the entity-physical module, only takes advantage of it if it exists.

**Comment by laf** [comment]
Yeah I didn't realise that until you pointed it out as I'd only read the reply on email where it said you rewritten it to pull from DB.

As you pointed out in irc, this still does mean that someone could have stale data if entity-physical was disabled 6 months ago as an example. It's not a huge concern but does still swing me more to why bother checking the DB if we can't be sure when that data was from.

**Comment by Rosiak** [comment]
As it's only disco job, I'm not concerned regarding the "extra" snmp req as this not that "expensive" at this point.

**Pull request merged**